File: 1678-goal-parser-interpretation/1678-goal-parser-interpretation.py
class Solution:
    def interpret(self, command: str) -> str:
        res =""
        c1= "()"
        c2="(al)"
        c=""
        for s in command:
            if s=="G":
                res+=s
            else:
                c+=s
                if(c==c1):
                    res+='o'
                    c=""
                elif(c==c2):
                    res+="al"
                    c=""
        return res

File: 1678-goal-parser-interpretation/1678-goal-parser-interpretation.go
func interpret(command string) string {
	result := ""
	for i := 0; i < len(command); i++ {
		if command[i] == 'G' {
			result += "G"
		} else if command[i] == '(' {
			if command[i+1] == ')' {
				result += "o"
				i++
			} else if command[i+1] == 'a' {
				result += "al"
				i += 3
			}
		}
	}
	return result
}